atlp-rwanda/e-commerce-mavericcks-bn
# ft-2FA two factor authentication for seller

## Purpose

The goal of implementing two-factor authentication (2FA) for sellers is to enhance the security measures within the system.

## Changes Made

- Modified the authentication controller and user controller.
- Created a new migration file for OTP (one-time password) functionality.
- Added a new column named "enableOTP" in the user table.
- Updated user and authentication routes.
- Made necessary adjustments in the Swagger documentation.
- Create a new endpoint to enable user two factor authentication

## Testing Instructions

To ensure proper functionality, follow these steps:

1. Run the migration script using npm run migration.
2. Create a seller user.
3. Enable 2FA for the user.
4. Log in as a seller.
5. Receive the OTP via email and token in postman.
6. Paste token where it's necessary in params based on a swagger docs and OTP as body req
7. Hint Send button


## Related Issues

- None

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:
